### ISSUE TITLE
[Discussion] choice of acceleration architecture

### ISSUE THREAD

**Issue by se4u** [xororz/local-dream#57]
Hi, I was curious what you think about the different acceleration paths available for accelerating gen-ai models, basically if I want to go from huggingface to accelerated app then I can either

1. convert to tensorflow and use TFLite and use TFlite's qualcomm HTP delegates,
2. or convert to onnx, and then use onnx runtime which also has qualcomm delegates,
3. or use pytorch mobile and executorch which also has qualcomm delagates
4. use qnn sdk directly.

I was wondering if you could please describe your reasons for choosing the fourth approach instead of using something more higher level, is the native qnn sdk much better supported or has much better performance?

**Comment by xororz** [comment]
I'm aware of the existence of these delegates, but I haven't used them personally.
- From what I understand, the quantization support in TFLite, ONNX, and PyTorch is limited to 8-bit activations. However, QNN can achieve 16-bit quantization, which is very helpful for model accuracy.
- Additionally, using higher-level delegates doesn't perform as well as using QNN directly. After model conversion, QNN performs very aggressive graph optimizations that are specifically tailored to the device model. Using QNN directly provides the fastest execution speed.
